chapters/data-analysis.tex

```diff
@@ -1,292 +1,488 @@
+If every script that creates a table starts by loading a data set, subsetting it and manipulating variables, any edits to construction need to be replicated in all scripts. 
+This increases the chances that at least one of them will have a different sample or variable definition.
+
+
+% What to do during construction -----------------------------------------
+Construction is the step where you face the largest risk of making a mistake that will affect your results. 
+Keep in mind that details and scales matter. 
+It is important to check and double-check the value-assignments of questions and their scales before constructing new variables using them. 
+Are they in percentages or proportions? 
+Are all variables you are combining into an index or average using the same scale? 
+Are yes or no questions coded as 0 and 1, or 1 and 2?
+This is when you will use the knowledge of the data you acquired and the documentation you created during the cleaning step the most.
+It is often useful to start looking at comparisons and other documentation 
+outside the code editor.
+
+Adding comments to the code explaining what you are doing and why is crucial here.
+There are always ways for things to go wrong that you never anticipated, but two issues to pay extra attention to are missing values and dropped observations. 
+If you are subsetting a data set, you should drop observations explicitly, indicating why you are doing that and how the data set changed.
+Merging, reshaping and aggregating data sets can change both the total number of observations and the number of observations with missing values.
+Make sure to read about how each command treats missing observations and, whenever possible, add automated checks in the script that throw an error message if the result is changing.
+
+At this point, you will also need to address some of the issues in the data that you identified during data cleaning. 
+The most common of them is the presence of outliers.
+How to treat outliers is a research question, but make sure to note what we the decision made by the research team, and how you came to it. 
+Results can be sensitive to the treatment of outliers, so keeping the original variable in the data set will allow you to test how much it affects the estimates.
+More generally, create derived measures in new variables instead of overwriting the original information.
+
+% Outputs -----------------------------------------------------------------
+
+% Data set
+The outputs of construction are the data sets that will be used for analysis.
+The level of observation of a constructed data set is the unit analysis. 
+Each data set is purpose-built to answer an analysis question.
+Since different pieces of analysis may require different samples,
+or even different units of observation,
+you may have one or multiple constructed data sets, 
+depending on how your analysis is structured.
```

Comment:
_Since different pieces of analysis may require different samples, or even different units of observation, you may have one or multiple constructed data sets, depending on how your analysis is structured._ 

I think we need to explain this better to someone who has never though of this. As in how does it depend on how the analysis is structured? How many data sets if multiple? I think we are explaining an advanced workflow in quite a vague way

Reply:
"Build the datasets you need, and not the ones you don't" ;-P

I think the key idea is that one project, or even one paper, is not one dataset. I'm not sure how to get that across exactly...

Reply:
addressed in 342f6b8